velox/functions/lib/MapFromEntries.cpp

```diff
@@ -140,9 +144,14 @@ class MapFromEntriesFunction : public exec::VectorFunction {
         for (auto i = 0; i < size; ++i) {
           // Check nulls in the top level row vector.
           const bool isMapEntryNull = decodedRowVector->isNullAt(offset + i);
-          if (isMapEntryNull) {
-            // Set the sizes to 0 so that the final map vector generated is
-            // valid in case we are inside a try. The map vector needs to be
+          if (isMapEntryNull && allowNullEle) {
```

Comment:
`allowNullEle` is not a local variable, maybe use `if constexpr (allowNullEle)` to create different branches for optimization.